gcs/COOPERATIVE_LOCKING.md

```diff
@@ -0,0 +1,106 @@
+### Introducing GCS Connector Cooperative Locking
+
+We are excited to release Cooperative Locking support for GCS Connector. With
+this release you will be able to isolate and recover directory modification
+operations performed through hadoop fs command line interface on Cloud Storage.
+
+The key motivation behind this feature is a fact that directory mutations are
+not atomic on GCS, which means that competing directory mutations are
+susceptible to race conditions that can lead to data integrity issues (loss,
+duplication, etc).
+
```

Comment:
These benefits are hard to understand in this context, I literally don't know what "write-only isolation", "recovery of failed directory operations" mean. Why do I want to isolation, why directory operations could fail and I need to care.

Reply:
Done.